src/cljs/harja/tiedot/kanavat/urakka/liikenne.cljs

```diff
@@ -169,41 +169,37 @@
 (defn koko-tapahtuma [rivi {:keys [haetut-tapahtumat]}]
   (some #(when (= (::lt/id rivi) (::lt/id %)) %) haetut-tapahtumat))
 
-(defn hae-sulutuksen-suunta [tapahtuma haluttu-toiminto haluttu-suunta]
-  ;; Palauttaa integerin montako suuntaa tapahtumalla 
+(defn hae-sulutuksen-suunta-lkm [tapahtuma haluttu-suunta]
+  ;; Palauttaa integerin montako kertaa haluttu suunta esiintyy tapahtumalla
```

Comment:
Miksi funktion nimeen lisätty -lkm? Lkm viittaa itsepalveluiden lukumäärään. Minusta tuo hae-sulutuksen-suunta on aivan hyvä nimi funktiolle, sillä sitä tässä juuri tehdään. Muuten OK.

Reply:
hae-sulutuksen-suunta antaisi ymmärtää että se palauttaisi :ylos tai :alas.
Tuo funktio palauttaa count funktion palauttaman lukumäärän, funktion nimestäkin voisi olla hyvä voida päätellä mitä se tekee.

Reply:
Aaaaa, eli tarkoitat siis hae-sulutuksen-suunta-lukumaara, aivan.. Olet oikeassa, tuo -lkm nimi on loogisempi.

Mutta tämän funktion voisi muuttaa palauttamaan booleanin, ja nimetä esim "onko-sulutuksella-suunta?", olisiko näin parempi?

Reply:
tuota funktiota joka nyt palauttaa lukumäärän montako kertaa sulutus tapahtumalla on annettu suunta on käytetty kolme kertaa.
- laske-yhteenveto tarkistaa onko positiivinen luku: (pos? (hae-sulutuksen-suunta-lkm tapahtuma haluttu-suunta))
- kasittele-suunta-alukselle näyttää tekevän samanlaisen tarkistuksen: sulutus-ylos? (> (hae-sulutuksen-suunta-lkm tapahtuma :ylos) 0)
- ja :alas samalla tavalla.

Kyllä se voisi palauttaa booleanin onko sillä parametrina annettua suuntaa.
"onko-sulutuksella-haluttu-suunta"?

Reply:
Käyttöliittymällä voi luoda sulutustapahtuman jolle ei anna suuntaa. Tiedä sitten onko tuo mielekästä. Tuossa tapauksessa "onko-sulutuksella-suunta?" voisi palauttaa false/nil. Jos sulutukselle on antanut suunnan (ylös taikka alas), kysyttäessä "onko-sulutuksella-suunta?" looginen vastaus olisi true :-)